MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2196,6 +2311,63 @@ - (void)processQueuedEvents:(void (^)(NSUInteger addedHistoryCellNb, NSUInteger
                         }
                     }
                 }
+
+                // Check if all cells of self.bubbles belongs to a single collapse serie.
+                // In this case, collapsableSerieAtStart and collapsableSerieAtEnd must be equal
+                // in order to handle next forward or backward pagination.
+                if (collapsableSerieAtStart == bubbles.firstObject)
+                {
+                    // Find the tail
+                    id<MXKRoomBubbleCellDataStoring> tailBubbleData = collapsableSerieAtStart;
+                    while (tailBubbleData.nextCollapsableCellData)
+                    {
+                        tailBubbleData = tailBubbleData.nextCollapsableCellData;
+                    }
+
+                    if (tailBubbleData == bubbles.lastObject)
+                    {
+                        collapsableSerieAtEnd = collapsableSerieAtStart;
+                    }
+                }
+                else if (collapsableSerieAtEnd)
+                {
+                    // Find the start
+                    id<MXKRoomBubbleCellDataStoring> startBubbleData = collapsableSerieAtEnd;
+                    while (startBubbleData.prevCollapsableCellData)
+                    {
+                        startBubbleData = startBubbleData.prevCollapsableCellData;
+                    }
+
+                    if (startBubbleData == bubbles.firstObject)
+                    {
+                        collapsableSerieAtStart = collapsableSerieAtEnd;
+                    }
+                }
+
+                // Compose (= compute collapsedAttributedTextMessage) of collapsable series
+                for (id<MXKRoomBubbleCellDataStoring> bubbleData in collapsingCellDataSeries)
+                {
+                    // Get all events of the serie
+                    NSMutableArray<MXEvent*> *events = [NSMutableArray array];
+                    id<MXKRoomBubbleCellDataStoring> nextBubbleData = bubbleData;
+                    do
+                    {
+                        [events addObjectsFromArray:nextBubbleData.events];
+                    }
+                    while ((nextBubbleData = nextBubbleData.nextCollapsableCellData));
+
+                    // Build the summary string for the serie
+                    bubbleData.collapsedAttributedTextMessage = [self.eventFormatter attributedStringFromEvents:events withRoomState:bubbleData.collapseState error:nil];
+
+                    // Release collapseState objects, even the one of collapsableSerieAtStart.
+                    // We do not need to keep its state because if an collapsable event comes before collapsableSerieAtStart,
+                    // we will take the room state of this event.
+                    if (bubbleData != collapsableSerieAtEnd &&
```

Comment:
"bubbleData != collapsableSerieAtEnd" is always true when "(bubbleData == collapsableSerieAtStart && collapsableSerieAtStart != collapsableSerieAtEnd)" is satisfied.